chronos/internal/asyncengine.nim

```diff
@@ -10,6 +10,94 @@
 
 {.push raises: [].}
 
+## This module implements the core asynchronous engine / dispatcher.
+##
+## The dispatcher acts as a kind of loop that wires events coming from the
+## operating system to tasks scheduled by the program. You must call
+## ``poll`` on it (or a function which does so for you such as ``waitFor`` or
+## ``runForever``) in order to poll for any outstanding events.
+##
+## The underlying implementation is based on epoll on Linux, IO Completion Ports
+## on Windows and select on other operating systems.
+##
+## The ``poll`` function will not, on its own, return any events. Instead
+## an appropriate ``Future`` object will be completed. A ``Future`` is a
+## type which holds a value which is not yet available, but which *may* be
```

Comment:
I'm not sure about *may* here. Of course its possible to create a Future which will never changes its state, but its rather edge case, so probably *may* is not a proper word here.

Reply:
81a6584a44c69c03c47117c848dbbe63fac79fb5